feat(developer guide): New documentation for CI/CD (#125)
kestra-io/kestra.io: docs/developer-guide/cicd/gitlab/README.md

---
order: 2
---

# GitLab

## GitLab CI

> GitLab integrate a solution call [GitLAb CI/CD](https://docs.gitlab.com/ee/ci/) that allow you to

Reviewer: ```suggestion
GitLab integrates a solution called [GitLab CI/CD](https://docs.gitlab.com/ee/ci/) that allows you to
```

> define pipelines in `YAML` files to automate tests, compilation or even deployment for your

Reviewer: ```suggestion
define pipelines in `YAML` files to automate tests, compilation or even deploy your
```

applications.

## Example

> Here is an example of a GitLab CI pipeline. We define 3 stages, `test` where we validate our flow,

Reviewer: ```suggestion
Here is an example of a GitLab CI pipeline. We define 3 stages, `test` where we validate our flows,
```

> `deploy-template` where we deploy our `templates` before our `flows`, to avoid running before their

Reviewer: ```suggestion
`deploy-template` where we deploy our `templates` before our `flows`, to avoid running flows before their
```

templates are created. And finally the stage `deploy-flow` where we deploy our flows.

```yaml
stages:
  - test
  - deploy-template
  - deploy-flow

default:
  image: kestra/kestra:latest

variables:
  KESTRA_HOST: https://kestra.io/

# --------------------------------------
# Validating Kestra Resources
# --------------------------------------
kestra-validate-flows-job:
  stage: test
  # Validate our flows on server-side
  script:
    - /app/kestra flow validate --server https://${KESTRA_HOST}/

kestra-validate-template-job:
  stage: test

>   # Validate our template within the client

Reviewer: ```suggestion
  # Validate our templates with the client
```

---
order: 2
---

# GitLab

## GitLab CI

> GitLab integrates a solution called [GitLAb CI/CD](https://docs.gitlab.com/ee/ci/) that allow you to

Reviewer 2: ```suggestion
GitLab integrates a solution called [GitLAb CI/CD](https://docs.gitlab.com/ee/ci/) that allows you to define pipelines in YAML files to automate tests, compilation, or even deployment for your applications.
```

> define pipelines in `YAML` files to automate tests, compilation, or even deploy for your

Reviewer 2: ```suggestion
```

> Applications.

Reviewer 2: ```suggestion
```

## Example

> Here is an example of a GitLab CI pipeline. We define 3 stages, `test` where we validate our flows,

Reviewer 2: ```suggestion
Here is an example of a GitLab CI pipeline. We define three stages:
*`test`, where we validate our flows 
*`deploy-template`, where we deploy our `templates`
* stage `deploy-flow`, where we deploy our flows.
```

> `deploy-template` where we deploy our `templates` and finally the stage `deploy-flow` where we deploy our flows.

Reviewer 2: ```suggestion
```

::: warning

> `Flows` should always be deployed before `Templates`, to avoid flows running before their templates are created.

Reviewer 2: ```suggestion
Flows should always be deployed before Templates to avoid flows running before their templates are created.
```
